### ISSUE TITLE
Fix `unit-no-unknown` false positives for `*vi` & `*vb` viewport units

### ISSUE THREAD

**Issue by sidverma32** [stylelint/stylelint#6427]
### What is the problem you're trying to solve?

As of March 2022 new CSS viewports units svh,lvh,dvh,svw,lvw and dvw are incoming as part of the efforts of Apple, Boucoup, Google, Igalia, Microsoft and Mozilla to imprve the web for developers.

Currently running style lint on "dvh" unit is breaking  event after adding to the "unit-whitelist" array.

### What solution would you like to see?

Support for this: svh,lvh,dvh,svw,lvw and dvw

**Comment by jeddy3** [comment]
@sidverma32 Thanks for the report.

Support for the [new range of viewport units](https://drafts.csswg.org/css-values-4/#viewport-relative-lengths) was added to the [`unit-no-unknown`](https://stylelint.io/user-guide/rules/unit-no-unknown) rule in [`stylelint@14.6.1`](https://stylelint.io/CHANGELOG#1461).

Our reference data is:

https://github.com/stylelint/stylelint/blob/142e4773223074ec6a6f9687d7f7351f0e9efcbb/lib/reference/units.js#L19-L36

Are we missing any?

> Currently running style lint on "dvh" unit is breaking event after adding to the "unit-whitelist" array.

Can you expand on this, please? The [`unit-allowed-list`](https://stylelint.io/user-guide/rules/unit-allowed-list) (previously called `unit-whitelist` back in version 13) should allow any unit to be listed and should be a breaking event.

**Comment by alexander-akait** [comment]
@jeddy3 Yeah, we miss some:
```
 //  Viewport-percentage Lengths
"vw" | "svw" | "lvw" | "dvw" |
"vh" | "svh" | "lvh" | "dvh" |
"vi" | "svi" | "lvi" | "dvi" |
"vb" | "svb" | "lvb" | "dvb" |
"vmin" | "svmin" | "lvmin" | "dvmin" |
"vmax" | "svmax" | "lvmax" | "dvmax"
```

**Comment by jeddy3** [comment]
@alexander-akait Thanks! It looks like we're missing the [`*vi`](https://drafts.csswg.org/css-values-4/#valdef-length-vi) and [`*vb`](https://drafts.csswg.org/css-values-4/#valdef-length-vb) units.

@sidverma32 I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time. It'll involve updating [our reference file](https://github.com/stylelint/stylelint/blob/142e4773223074ec6a6f9687d7f7351f0e9efcbb/lib/reference/units.js#L19-L36).

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by sidverma32** [comment]
Sure will pick this up! @jeddy3  please assign this to me

**Comment by sidverma32** [comment]
I have added the missing units in the reference file , Please have a look over it.

Also, I haven't added example in [test](lib/rules/unit-no-unknown/__tests__/index.js) file as `*vi` and `*vb` are not written there. Do let me know if we have to add in the same PR or different PR.

**Comment by ybiquitous** [comment]
@sidverma32 Thanks for creating the pull request. I don't think we need to add test cases because the change is explicit.
